_internal/utils/mutate.ts

```diff
@@ -54,16 +54,20 @@ export async function internalMutate<Data>(
   const revalidate = options.revalidate !== false
   const rollbackOnError = options.rollbackOnError !== false
 
-  // If 2nd arg is key filter, return the mutation results of filtered keys
+  // If the second argument is a key filter, return the mutation results for all
+  // filtered keys.
   if (isFunction(_key)) {
     const keyFilter = _key
     const matchedKeys: Key[] = []
-    for (const state of cache.values()) {
-      if (keyFilter(state.key)) matchedKeys.push(state.key)
+    for (const key of cache.keys()) {
+      if (keyFilter((cache.get(key) as { _k: any })._k)) {
```

Comment:
```js
function bench() {
  const map = new Map()
  for (let i = 0; i < 10000000; i++) {
    map.set(i, i)
  }

  const label1 = 'keys + get'
  console.time(label1)
  for (const k of map.keys()) {
    map.get(k)
  }
  console.timeEnd(label1)

  const label2 = 'values'
  console.time(label2)
  for (const v of map.values()) {
    v
  }
  console.timeEnd(label2)
}
```

`keys()` + `get` will be slower than values 🤔 
```
keys + get: 216.172ms
values: 79.717ms
```

Reply:
It's too unrealistic to have 10 million keys in the cache though. Your app will be out of memory before that happens 😄

Reply:
Yeah right, let's keep `keys` + `get()` for orthogonality